### ISSUE TITLE
05_run_models.sh fails without GPU

### ISSUE THREAD

**Issue by meowcat** [samgoldman97/mist#1]
Note that cudatoolkit is already commented out in environment.yml as of 2c4c801.

Dockerfile:
```
FROM mambaorg/micromamba:latest

COPY --chown=$MAMBA_USER:$MAMBA_USER mist/environment.yml /tmp/env.yml
RUN micromamba install -y -n base -f /tmp/env.yml && \
    micromamba clean --all --yes

RUN micromamba install -y -n base -c conda-forge \
    git curl wget unzip

COPY --chown=$MAMBA_USER:$MAMBA_USER mist /mist
ARG MAMBA_DOCKERFILE_ACTIVATE=1
RUN pip install -q --exists-action i -r /mist/requirements.txt && \
    cd /mist && \
    wget -q https://bio.informatik.uni-jena.de/repository/dist-release-local/de/unijena/bioinf/ms/sirius/4.9.3/sirius-4.9.3-linux64-headless.zip && \
    unzip -q sirius-4.9.3-linux64-headless.zip && \
    rm sirius-4.9.3-linux64-headless.zip && \
    python setup.py develop
```

Building:
```
git clone git@github.com:samgoldman97/mist.git
docker build -t mist .
```

```
source quickstart/00_download_models.sh
# works
python quickstart/01_reformat_mgf.py
# works (except mkdir: cannot create directory ‘data/paired_spectra/quickstart’: File exists)
source quickstart/02_run_sirius.sh
# works
python quickstart/03_summarize_sirius.py
# works
python quickstart/04_create_lookup.py
# works
source quickstart/05_run_models.sh
# gives following error:
```

```
mkdir: cannot create directory ‘quickstart/model_predictions’: File exists
Traceback (most recent call last):
  File "run_scripts/pred_fp.py", line 8, in <module>
    pred_fp.run_fp_pred()
  File "/mist/src/mist/pred_fp.py", line 70, in run_fp_pred
    pretrain_ckpt = torch.load(model_ckpt)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 607, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 882, in _load
    result = unpickler.load()
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 857, in persistent_load
    load_tensor(data_type, size, key, _maybe_decode_ascii(location))
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 846, in load_tensor
    loaded_storages[key] = restore_location(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 175, in default_restore_location
    result = fn(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 151, in _cuda_deserialize
    device = validate_cuda_device(location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 135, in validate_cuda_device
    raise RuntimeError('Attempting to deserialize object on a CUDA '
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
Traceback (most recent call last):
  File "run_scripts/retrieval_contrastive.py", line 9, in <module>
    retrieval_contrast.run_contrastive_retrieval()
  File "/mist/src/mist/retrieval_contrast.py", line 123, in run_contrastive_retrieval
    pretrain_ckpt = torch.load(model_ckpt)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 607, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 882, in _load
    result = unpickler.load()
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 857, in persistent_load
    load_tensor(data_type, size, key, _maybe_decode_ascii(location))
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 846, in load_tensor
    loaded_storages[key] = restore_location(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 175, in default_restore_location
    result = fn(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 151, in _cuda_deserialize
    device = validate_cuda_device(location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 135, in validate_cuda_device
    raise RuntimeError('Attempting to deserialize object on a CUDA '
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
100%|███████████████████████████████████████████| 7/7 [00:00<00:00, 2498.73it/s]
Traceback (most recent call last):
  File "run_scripts/embed_contrastive.py", line 9, in <module>
    embed_contrast.embed_specs()
  File "/mist/src/mist/embed_contrast.py", line 62, in embed_specs
    pretrain_ckpt = torch.load(model_ckpt)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 607, in load
    return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 882, in _load
    result = unpickler.load()
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 857, in persistent_load
    load_tensor(data_type, size, key, _maybe_decode_ascii(location))
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 846, in load_tensor
    loaded_storages[key] = restore_location(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 175, in default_restore_location
    result = fn(storage, location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 151, in _cuda_deserialize
    device = validate_cuda_device(location)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 135, in validate_cuda_device
    raise RuntimeError('Attempting to deserialize object on a CUDA '
RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
```

I have not yet tried adding `--gpu false` parameters into the quickstart scripts, which seems like it would do something in `pred_fp` and `retrieval_contrastive`.

**Comment by samgoldman97** [comment]
Thanks for pointing this out! I believe this errors because I saved the models to gpu when I exported them, and so they attempt to load directly on gpu. As mentioned in the traceback, it looks like adding `map_location=torch.device('cpu')` into each torch.load call should correct this. The `--gpu` flag is set to false by default, so this shouldn't be an issue. 

I've pushed the requisite changes. Let me know if you still have errors.
